### PR TITLE
Rename jobs to hopefully more descriptive titles

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Rename job titles to make jobs more descriptive:
- Remove `run_` prefixes in jobs.
- Rename remaining `build_*` jobs to `docker_build_*` for docker jobs.
- Rename `windows_msi_x64-a7` to `windows_msi_and_bosh_zip_x64-a7` to better describe its output.

### Motivation

Make pipeline easier to view

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -2007,7 +2007,7 @@ testkitchen_cleanup_azure-a7:
     ARCH: arm64
 
 # build agent6 py2 image
-build_agent6:
+docker_agent6:
```
Comment (albertvaka) [inline .gitlab-ci.yml]: 🍰

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -2124,15 +2124,15 @@ build_agent7_jmx_arm64:
     TESTING_ARG:  --target testing --build-arg WITH_JMX=true --build-arg PYTHON_VERSION=3 --build-arg DD_AGENT_ARTIFACT=datadog-agent_7*_arm64.deb
 
 # build the cluster-agent image
-build_cluster_agent_amd64:
+cluster_agent_image_amd64:
```
Comment (albertvaka) [inline .gitlab-ci.yml]: `docker_cluster_agent`? For consistency with the other docker image build jobs.

**Review by albertvaka — COMMENTED** [review]
Should the `test_`, `docker_`, etc. prefixes become suffixes (eg: `rpm-x64-py3-test`)? So the most relevant info is first in the name.

Also, the omnibus build jobs could also be identified the same way with a `-omnibus` or `-package` or something like that. Specially when listing those jobs as dependencies of others, it's weird that they are just called `agent_suse-x64-a7`.

File: .gitlab-ci.yml (hunk 2 above)
Comment (albertvaka) [reply .gitlab-ci.yml]: Done at merge time.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -649,7 +649,7 @@ run_dogstatsd_size_test:
     - inv -e dogstatsd.size-test --skip-build
 
 # check the size of the static dogstatsd binary
-run_dogstatsd_arm_size_test:
+dogstatsd_arm_size_test:
```
Comment (KSerrania) [inline .gitlab-ci.yml:652]:
```suggestion
dogstatsd_arm64_size_test:
```

to be in line with the other arm64 jobs?

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -276,13 +276,13 @@ linux_armhf_go_deps:
   variables:
     NEW_BUILDER: "true"
 
-run_tests_windows-x64:
+tests_windows-x64:
   extends: .run_tests_windows_base
```
Comment (KSerrania) [inline .gitlab-ci.yml:280]: Nit: the base job could also be renamed

**Review by KSerrania — APPROVED** [review]
LGTM

**Pull request merged**